example_client/grpc_stateful_client.py

```diff
@@ -26,160 +26,356 @@
+    print('Context window right width cw_r: {}'.format(cw_r))
+    sequence_id = int(args.get('sequence_id'))
+    print('Starting sequence_id: {}'.format(sequence_id))
+    print('Start processing:')
+    print('Model name: {}'.format(args.get('model_name')))
+
+    data_iterator, input_names, output_names, input_data, reference_scores = prepare_processing_data(
+        args)
+
+    # Input control tokens
+    SEQUENCE_START = 1
+    SEQUENCE_END = 2
+
+    global_avg_rms_error_sum = 0.0
+
+    # Main inference loop
+    for key in data_iterator:
+        obj = data_iterator[key]
+        sequence_size = obj.shape[0]
+        print('\n\tInput name: {}'.format(key))
+        print_debug('\tInput in shape: {}'.format(obj.shape))
+        print_debug('\tInput sequence size: {}'.format(sequence_size))
+        print_debug('\tSequence id: {}'.format(sequence_id))
+
+        if sequence_size == 1:
+            print('\nERROR: Sequence size equal {} in input data must be bigger than 1.'.format(
+                sequence_size))
             exit(1)
 
-        duration = (end_time - start_time).total_seconds() * 1000
-        processing_times = np.append(processing_times,np.array([int(duration)]))
-        resultsArray = make_ndarray(result.outputs[args['output_name']])
-
-        # Reset sequence end for testing purpose
-        #request.inputs['sequence_control_input'].CopyFrom(make_tensor_proto(SEQUENCE_END, dtype="uint32"))
-        #result = stub.Predict(request, 10.0) # result includes a dictionary with all model outputs
-        
-        #resultsArray = np.array(output)
-        referenceArray = scoreObjects[key][x]
-        
-        meanErr = CalculateUtteranceError(referenceArray, resultsArray[0])
-
-        meanErrSum += meanErr
-        # Statistics
-        printDebug('Iteration {}; Mean error: {:.10f} Processing time: {:.2f} ms; speed {:.2f} fps\n'.format(x,meanErr,round(np.average(duration), 2), round(1000 * batch_size / np.average(duration), 2)))
-
-        #END utterance loop
-
-    meanErrAvg = meanErrSum/min(samples_limit,batch_size)
-    print("\tSequence {} mean error: {:.10f}\n".format(sequence_id, meanErrAvg))
-    sequence_id += 1
-    meanErrGlobal += meanErrAvg
-    #END input name loop
-
-meanGlobalErrAvg = meanErrGlobal/min(numberOfKeys, utterances_limit)
-print("Global mean error: {:.10f}\n".format(meanGlobalErrAvg))
-print_statistics(processing_times, batch_size)
+        mean_avg_rms_error_sum = 0.0
+        score_index = (cw_l + cw_r) * -1
+
+        for x in range(0, sequence_size + cw_l + cw_r):
+            print_debug('\tExecution: {}\n'.format(x))
+            request = predict_pb2.PredictRequest()
+            request.model_spec.name = args.get('model_name')
+
+            # Set input data index
+            input_index = x
+            # Input for context window
+            if x < cw_l:
+                # Repeating first input data infer request cw_l times for the
```

Comment:
Maybe instead of "first input data infer request" we could go with "first frame"?